sdk/communication/communication-common/src/credential/clientArguments.ts

```diff
@@ -27,8 +27,14 @@ const assertValidEndpoint = (host: string): void => {
  *
  * @param credential - The credential being checked.
  */
-export const isKeyCredential = (credential: any): credential is KeyCredential => {
-  return credential && typeof credential.key === "string" && credential.getToken === undefined;
+export const isKeyCredential = (
+  credential?: KeyCredential | PipelineOptions | TokenCredential
```

Comment:
This is odd. A type guard like this should accept all kind of input and return whether it satisfies a condition. Let's follow what `isTokenCredential` is doing and accept `unknown`, then test for the key property, its type being string and the absence of `getToken` to be safe.

https://github.com/Azure/azure-sdk-for-js/blob/8a863519e05fa4a3a0a80b6d016ae1c646e51b3b/sdk/core/core-auth/src/tokenCredential.ts#L72-L87

Reply:
Do we want to also update the parameters in `parseClientArguments` to unknown as well?